bindings/java/src/junit/com/apple/foundationdb/RangeQueryTest.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * RangeQueryTest.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.apple.foundationdb;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+import com.apple.foundationdb.async.AsyncIterable;
+import com.apple.foundationdb.tuple.FastByteComparisons;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * Tests around the Range Query logic.
+ * 
+ * These tests do _not_ require a running FDB server to function. Instead, we
+ * are operating on a good-faith "The underlying native library is correct"
+ * functionality. For end-to-end tests which require a running server, see the
+ * src/tests source folder.
+ */
+@RunWith(Parameterized.class)
+public class RangeQueryTest {
+    private static Executor EXECUTOR = new Executor() {
+
+        @Override
+        public void execute(Runnable command) {
+            command.run();
+        }
+    };
+    private StreamingMode mode;
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> params(){
+        return  Arrays.<Object[]>asList(
+            new Object[]{StreamingMode.WANT_ALL},
+            new Object[]{StreamingMode.ITERATOR},
+            new Object[]{StreamingMode.EXACT},
+            new Object[]{StreamingMode.SMALL},
+            new Object[]{StreamingMode.MEDIUM},
+            new Object[]{StreamingMode.LARGE},
+            new Object[]{StreamingMode.SERIAL}
+        );
+    }
+
+    public RangeQueryTest(StreamingMode mode) {
+        this.mode = mode;
+    }
+
+    private static FDBDatabase makeFakeDatabase(List<Map.Entry<byte[],byte[]>> data){
+        return new FDBDatabase(1, EXECUTOR) {
+            private long txnCounter = 3;
```

Comment:
nope. Completely arbitrary. I wanted it to be high enough that it was unlikely to have the same number as other objects that would be created, on the off chance that future development of the mock _did_ require unique transaction ids, but otherwise picked the number out of a hat.

Reply:
It would probably could to add a comment to that affect in the code.